### DISCUSSION TITLE
BBEdit support

### DISCUSSION THREAD

**Discussion by somelinguist** [artempyanykh/marksman#206] [Q&A] [answered]
Hi. I've been following development of Marksman for a while and finally decided to try it out...but in BBEdit. :) 

I followed the instructions for installing Marksman on Mac, and can start it from the command line.

I followed BBEdit's instructions for LSPs at https://www.barebones.com/support/bbedit/lsp-notes.html.

BBEdit can find and start the server, but it doesn't seem to work with it. BBEdit's log for the server reports the following every time it to request completion: 

`Error returned from LSP completion request: (null)`.

Is there any special configuration I might night need to pass to Marksman? Right now I've just setup the default configuration file from https://github.com/artempyanykh/marksman/blob/main/Tests/default.marksman.toml in `$HOME/.config/marksman/config.toml`.

Or is this more likely something specific to BBEdit?

Thanks!

**Comment by somelinguist** [comment, marked as answer]
Ok I think I figured it out. I read **more of the setup instructions** and added a `.marksman.toml` file to the directory where the markdown files were. Now it seems to work :)

**Reply by artempyanykh** [reply]
That's cool @somelinguist! Could you share the config for BBEdit - I'd be happy to edit it to the README setup instructions.

Side note - I'd expect marksman to turn on single file mode without .marksman.toml and still work fine. This needs to be looked at.

**Reply by somelinguist** [reply]
@artempyanykh After installing Marksman and making sure it was in my path, I edited the custom language settings for markdown in BBEdit as described at https://www.barebones.com/support/bbedit/lsp-notes.html#config. On the Server tab, I set `Command` to `marksman` and `Language ID` to `markdown`. After changing the command, BBEdit detected Marksman and started using it when I opened markdown files. I left the `Configuration` settings as `Default`.

As noted above, however, BBEdit's logs indicated Marksman was returning `null` for completion requests until I added the .marksman.toml file to the directory where the markdown files were located. 

BBEdit does provide a place to add arguments to the start up. Does Marksman need a specific parameter to set to enter single file mode, or should it do it automatically?

BBEdit also allows creating some more specific configuration options for a server, as described at https://www.barebones.com/support/bbedit/lsp-notes.html#advanced, but I don't completely understand all of the settings.
